test/miscellaneous/async-await.js

```diff
@@ -716,13 +717,15 @@ suite('async', () => {
     testParseFailure('(async function* (){})', 'Unexpected token "*"');
     testParseFailure('({ async *a(){} })', 'Unexpected token "*"');
     testParseFailure('async await => 0', '"await" may not be used as an identifier in this context');
-    testParseFailure('async (await) => 0', 'Async arrow parameters may not contain "await"');
+    testParseFailure('async (await) => 0', ErrorMessages.NO_AWAIT_IN_ASYNC_PARAMS);
     testParseFailure('(class { async })', 'Only methods are allowed in classes');
     testParseFailure('(class { async\na(){} })', 'Only methods are allowed in classes');
     testParseFailure('(class { async get a(){} })', 'Unexpected identifier');
-    testParseFailure('async (a = await => {}) => {}', 'Async arrow parameters may not contain "await"');
-    testParseFailure('async (a = (await) => {}) => {}', 'Async arrow parameters may not contain "await"');
-    testParseFailure('async (a = aw\\u{61}it => {}) => {}', 'Async arrow parameters may not contain "await"');
+    testParseFailure('async (a = await => {}) => {}', ErrorMessages.NO_AWAIT_IN_ASYNC_PARAMS);
+    testParseFailure('async (a = (await) => {}) => {}', ErrorMessages.NO_AWAIT_IN_ASYNC_PARAMS);
+    testParseFailure('async (a = aw\\u{61}it => {}) => {}', ErrorMessages.NO_AWAIT_IN_ASYNC_PARAMS);
     testParseFailure('async (a = (b = await (0)) => {}) => {}', 'Async arrow parameters may not contain "await"');
+    testParseFailure('async ({await}) => 1', ErrorMessages.NO_AWAIT_IN_ASYNC_PARAMS);
+    testParseFailure('async function x({await}) { return 1 }', ErrorMessages.ILLEGAL_AWAIT_IDENTIFIER);
```

Comment:
Can you add the test from [this comment](https://github.com/shapesecurity/shift-parser-js/issues/416#issuecomment-488487357) as well? The test is

```js
async function f() { return {await}; }
```

Reply:
Added